src/Public/OpenApi.ps1

```diff
@@ -1247,47 +1792,165 @@ An component schema name.
+
+.PARAMETER Deprecated
+If supplied, the schema will be treated as Deprecated where supported.
+
 .PARAMETER Array
-If supplied, the schema reference will be treated as an array.
+If supplied, the schema will be treated as an array of objects.
+
+.PARAMETER Nullable
+If supplied, the schema will be treated as Nullable.
+
+.PARAMETER ReadOnly
+If supplied, the schema will be included in a response but not in a request
+
+.PARAMETER WriteOnly
+If supplied, the schema will be included in a request but not in a response  
+
+.PARAMETER MinProperties
+If supplied, will restrict the minimun number of properties allowed in an schema.
+
+.PARAMETER MaxProperties
+If supplied, will restrict the maximum number of properties allowed in an schema.
+
+.PARAMETER Array
+If supplied, the schema will be treated as an array of objects.
+
+.PARAMETER UniqueItems
+If supplied, specify that all items in the array must be unique
+
+.PARAMETER MinItems
+If supplied, specify minimum length of an array 
+
+.PARAMETER MaxItems
+If supplied, specify maximum length of an array 
+
+.PARAMETER Xml
+If supplied, controls the XML serialization behavior
+
 
 .EXAMPLE
 New-PodeOASchemaProperty -Name 'Config' -ComponentSchema "MyConfigSchema"
 #>
 function New-PodeOASchemaProperty
 {
-    [CmdletBinding()]
+    [CmdletBinding(DefaultParameterSetName = 'Inbuilt')]
     param(
-        [Parameter()]
+        [Parameter(Mandatory = $true)]
         [string]
         $Name,
 
-        [Parameter(Mandatory=$true)]
+        [Parameter(Mandatory = $true)]
         [string]
-        $Reference,
+        $ComponentSchema, 
 
-        [Parameter()]
+        [Parameter(ParameterSetName = 'Array')] 
         [string]
         $Description,
 
+        [Parameter(ParameterSetName = 'Array')] 
+        [String]
+        $Example,
+
+        [Parameter(ParameterSetName = 'Array')]
         [switch]
-        $Array
-    )
+        $Deprecated, 
 
-    if(!(Test-PodeOAComponentSchema -Name $Reference)) {
-        throw "The OpenApi component schema doesn't exist: $($Reference)"
-    }
+        [Parameter(ParameterSetName = 'Array')]
+        [switch]
+        $Required,
+
+        [Parameter(ParameterSetName = 'Array')]
+        [switch]
+        $Nullable, 
+
+        [Parameter(ParameterSetName = 'Array')]
+        [switch]
+        $ReadOnly,
+
+        [Parameter(ParameterSetName = 'Array')]
+        [switch]
+        $WriteOnly,
+
+        [Parameter(ParameterSetName = 'Array')]
+        [int]
+        $MinProperties,
+
+        [Parameter(ParameterSetName = 'Array')]
+        [int]
+        $MaxProperties,
 
+        [Parameter(Mandatory = $true, ParameterSetName = 'Array')]
+        [switch]
+        $Array,
+
+        [Parameter(ParameterSetName = 'Array')]
+        [switch] 
+        $UniqueItems,
+
+        [Parameter(ParameterSetName = 'Array')]
+        [int]
+        $MinItems,
+
+        [Parameter(ParameterSetName = 'Array')]
+        [int]
+        $MaxItems, 
```

Comment:
How come all of these a restricted to the Array parameter set? Ie, before `-Description` was settable freely, which might mean for that parameter this is a breaking change.

Reply:
because it has no impact on the openAPI spec 
but I can remove the restriction. It doesn't create any harm

Reply:
-description is freely settable